src/main/core/worker.h

```diff
@@ -52,6 +52,22 @@ void workerpool_joinAll(WorkerPool* pool);
 void workerpool_free(WorkerPool* pool);
 pthread_t workerpool_getThread(WorkerPool* pool, int threadId);
 
+// Compute the global min event time across all workers. We dynamically compute
+// the minimum time that we'll need for the next event round as the minimum of
+// i.) all events pushed by all workers during this round, and
+// ii.) the next queued event for all worker at the point when they stop
+// executing events.
+//
+// This func is not thread safe, so only call from the scheduler thread when the
+// workers are idle.
+SimulationTime workerpool_getGlobalNextEventTime(WorkerPool* workerPool);
+
+// The worker either pushed an event or finished executing its events and is
+// reporting the min time of events in their event queue.
+// This function is thread-safe.
+void worker_setMinEventTimeNextRound(SimulationTime simtime,
+                                     SimulationTime currentRoundEndTime);
```

Comment:
Do we need to pass the currentRoundEndTime? Worker already has it, right?

Reply:
Right, I missed that! I don't think we need it after reverting to the linear scan.